src/ConnectedMode.UnitTests/Persistence/BindingDtoConverterTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarLint.VisualStudio.ConnectedMode.Persistence;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.TestInfrastructure;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Persistence;
+
+[TestClass]
+public class BindingDtoConverterTests
+{
+    [TestMethod]
+    public void MefCtor_CheckIsExported()
+    {
+        MefTestHelpers.CheckTypeCanBeImported<BindingDtoConverter, IBindingDtoConverter>();
+    }
+
+    [TestMethod]
+    public void MefCtor_CheckIsSingleton()
+    {
+        MefTestHelpers.CheckIsSingletonMefComponent<BindingDtoConverter>();
+    }
+    
+    [TestMethod]
+    public void ConvertFromDto_ConvertsCorrectly()
+    {
+        const string localBindingKey = "solution123";
+        var bindingDto = new BindingDto
+        {
+            ProjectKey = "project123",
+            Profiles = new Dictionary<Language, ApplicableQualityProfile>(),
+            ProjectName = "ignored",
+            Organization = new SonarQubeOrganization("ignored", "ignored"),
+            ServerUri = new Uri("http://ignored"),
+            ServerConnectionId = "ignored", // only used for connection extraction, delegates to the connection object during the conversion
+        };
+        var connection = new ServerConnection.SonarCloud("myorg");
+
+        var boundServerProject = new BindingDtoConverter().ConvertFromDto(bindingDto, connection, localBindingKey);
```

Comment:
Please extract all object creation of BindingDtoConverter into a variable in TestInitialize.